### PR TITLE
RavenDB-13984 - Load with Include doesn't work for a dictionary

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-13984

### CODE REVIEW

File: test/FastTests/Client/Includes.cs [hunk 1]
```diff
@@ -433,6 +434,84 @@ public void Can_Include_By_List_Of_Secondary_String_Property_Using_IIncludeBuild
...
+                    session.Load<User>(userId1);
+                    session.Load<User>(userId2);
+
+                    Assert.Equal(1, session.Advanced.NumberOfRequests);
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var test = session
+                        .Include<Test>(x => x.Dictionary.Values.Select(d => d.UserId))
```
Comment (ayende) [inline test/FastTests/Client/Includes.cs]: What about keys?

Comment (grisha-kotler) [reply test/FastTests/Client/Includes.cs]: added a test

File: src/Raven.Client/Extensions/ExpressionExtensions.cs [hunk 2]
```diff
@@ -98,16 +98,21 @@ public static string ToPropertyPath(this Expression expression, char propertySep
...
                     var nextVal = propertyPathExpressionVisitor.Results.Peek();
-
                     if (nextVal.Length == collectionSeparator.Length && nextVal.Equals(collectionSeparator, StringComparison.OrdinalIgnoreCase))
                     {
                         continue;
                     }
                 }
 
-                builder.Append(curValue);
+                if (curValue.Length == 6 && curValue == "$Value" && i != stackLength - 1)
```
Comment (garayx) [inline src/Raven.Client/Extensions/ExpressionExtensions.cs]: `curValue.Length == 6` is redundant, no? Already checked by the string comparer.

Comment (grisha-kotler) [reply src/Raven.Client/Extensions/ExpressionExtensions.cs]: done

**Review by ppekrol — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Pull request merged**